### DISCUSSION TITLE
When will React be upgraded to 18?

### DISCUSSION THREAD

**Discussion by zwl1619** [remix-run/remix#3056] [Q&A] [answered]
When will React be upgraded to 18?

**Comment by MichaelDeBoey** [comment, marked as answer]
Remix is already React v18 compatible.

Templates will be updated once https://github.com/remix-run/remix/pull/2540 is merged.

**Comment by ggascoigne** [comment]
I'd add that due to issues with browser dev tools injecting into the DOM, you can easily get a broken dev build if you use the new hydrateRoot.  Being new to remix it took me way too long to work out that this wasn't something that I'd done wrong.

see https://github.com/remix-run/remix/issues/2947 and https://github.com/facebook/react/issues/22833 for more details.

Right now I'm using this.

```ts
if (process.env.NODE_ENV !== 'production') {
  console.warn(
    'Using ReactDOM.hydrate rather than hydrateRoot until the latter is patched to not break when dev tools inject into the dom'
  )
  require('react-dom').hydrate(<RemixBrowser />, document)
} else {
  hydrateRoot(document, <RemixBrowser />)
}
```

Hopefully this will stop being an issue fairly soon.
